0025-bundle_hot_loading.md

```diff
@@ -0,0 +1,138 @@
+# ADR 0025: Hot-loading of Runtime Bundles
+
+## Component
+
+Oasis Core
+
+## Changelog
+
+- 2025-07-13: Initial version
+
+## Status
+
+Accepted
+
+## Context
+
+The secure procedure for upgrading runtimes is not simple and requires
+significant effort from node operators. First, they must download
+the latest bundle, which includes the updated version of the runtime
+they intend to upgrade. Then, they need to verify the bundle, update
+the node configuration to point to the new bundle's location, and restart
+the node. Finally, once the new version is active, they must remove
+the outdated bundles.
+
+## Decision
+
+This proposal aims to automate bundle discovery and distribution so that
+the upgrade process is more secure, reliable, and more user-friendly
+for node operators.
+
+The process would involve the following steps:
+
+- The runtime owner publishes the bundle checksum, i.e. the SHA-256 hash
+  of the runtime bundle manifest, on-chain when registering a new runtime
+  deployment.
+
+- Upon registration of the new deployment, the node automatically retrieves
+  the bundle URL corresponding to the provided checksum and downloads
+  the bundle. The downloaded bundle is then verified and extracted to
+  the appropriate location.
+
+- Once the new deployment becomes active, files and bundles associated
+  with the previous versions are removed from the file system.
+
+If needed, we can gradually transition to a more decentralized version
+of this process.
+
+## Bundle Registry
+
+The bundle registry is responsible for storing metadata files used for bundle
```

Comment:
Could you maybe add as an example the default link for `registries`, and what format is supported for listing metadata files?